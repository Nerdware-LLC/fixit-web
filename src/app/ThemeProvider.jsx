import React from "react";
import { ThemeProvider as MTP, createTheme } from "@mui/material/styles";
import CssBaseline from "@mui/material/CssBaseline";
import { any } from "../types";

export const ThemeProvider = ({ children }) => {
  const theme = createTheme(THEME);

  return (
    <MTP theme={theme}>
      <CssBaseline />
      {children}
    </MTP>
  );
};

const THEME = {
  description: "dark",
  palette: {
    mode: "dark",
    primary: {
      light: "#f89a35",
      main: "#f78103", // Fixit orange
      dark: "#ac5a02"
    },
    secondary: {
      light: "#5381ff",
      main: "#2962ff", // blue
      dark: "#1c44b2"
    },
    success: {
      light: "#33d373",
      main: "#00c851",
      dark: "#008c38"
    },
    error: {
      main: "rgba(255, 115, 115, 0.87)"
    },
    info: {
      main: "#ff80ab" // AKA NOTIFICATION
    },
    text: {
      primary: "rgba(255, 255, 255, 0.87)",
      disabled: "rgba(255, 255, 255, 0.38)",
      hint: "rgba(255, 255, 255, 0.54)", // AKA PLACEHOLDER
      labels: "#f7a44a" // desaturated Fixit orange (CUSTOM KEY)
    },
    background: {
      paper: "#353535", // AKA SURFACE
      default: "#121212",
      backdrop: "rgba(0, 0, 0, 0.5)" // (CUSTOM KEY)
    }
  }
};

ThemeProvider.propTypes = {
  children: any.isRequired
};
