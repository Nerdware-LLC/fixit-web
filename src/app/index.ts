export { App } from "./App";

// Also export apollo-cache reactive vars:
export * from "./apolloCache/reactiveVars";
