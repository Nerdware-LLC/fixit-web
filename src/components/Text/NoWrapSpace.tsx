/**
 * Use this component to add a non-breaking space that will not wrap to a new line.
 *
 * ```tsx
 * // Example:
 * <p>
 *   This text will wrap as needed, but the last word will "stick" to the HelpInfoIcon
 *   <NoWrapSpace />
 *   <HelpInfoIcon
 *     tooltip="This icon will never be orphaned onto its own line all alone - huzzah!ðŸŽ‰"
 *   />
 * </p>
 * ```
 */
export const NoWrapSpace = () => (
  <span
    style={{
      fontSize: "inherit",
      fontWeight: "inherit",
      lineHeight: "inherit",
      whiteSpace: "nowrap",
    }}
  >
    &nbsp;
  </span>
);
