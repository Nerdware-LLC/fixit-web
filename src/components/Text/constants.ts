/**
 * Non-breaking space character. @docs https://unicodeplus.com/U+00a0
 */
export const NON_BREAKING_SPACE_CHAR = "\u00a0";

/**
 * Non-breaking hyphen character. @docs https://unicodeplus.com/U+2011
 */
export const NON_BREAKING_HYPHEN_CHAR = "\u2011";
