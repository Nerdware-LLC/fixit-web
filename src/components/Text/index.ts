export * from "./ClickToCopyText";
export * from "./NoWrapSpace";
