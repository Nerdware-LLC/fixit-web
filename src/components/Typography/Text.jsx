import React from "react";
import styled from "styled-components";
import { node } from "../../types";

export const Text = ({ children, ...props }) => (
  <StyledText {...props}>{children}</StyledText>
);

const StyledText = styled.p`
  font-size: 1rem;
  margin: 0;
  color: ${props => props.theme.palette.text.primary};
  line-height: 1rem;
`;

Text.propTypes = {
  children: node.isRequired
};
