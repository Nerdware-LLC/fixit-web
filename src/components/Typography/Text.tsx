import styled from "@emotion/styled";
import { useTheme } from "@mui/material/styles";
import { node } from "@types";

export const Text = ({
  variant = "primary",
  style = {},
  children,
  ...props
}: React.ComponentPropsWithoutRef<"p"> & { variant?: "primary" | "secondary" | "disabled" }) => {
  const theme = useTheme();

  return (
    <StyledText style={{ color: theme.palette.text[variant], ...style }} {...props}>
      {children}
    </StyledText>
  );
};

const StyledText = styled.p`
  font-family: Roboto, sans-serif;
  font-size: 1rem;
  margin: 0;
  line-height: 1.35rem;
`;

Text.propTypes = {
  children: node.isRequired
};
