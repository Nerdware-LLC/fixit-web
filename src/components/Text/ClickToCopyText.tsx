import { styled, alpha } from "@mui/material/styles";
import ButtonBase, { buttonBaseClasses } from "@mui/material/ButtonBase";
import IconButton, { iconButtonClasses } from "@mui/material/IconButton";
import { svgIconClasses } from "@mui/material/SvgIcon";
import Tooltip from "@mui/material/Tooltip";
import ContentCopyIcon from "@mui/icons-material/ContentCopy";
import { THEMES } from "@/app/ThemeProvider/themes";

export const ClickToCopyText = ({
  children: textToCopy,
  stripeWhitespace = false,
  tooltipTitle = "Click to copy",
}: ClickToCopyTextProps) => {
  const handleClickCopyBtn = () =>
    window.navigator.clipboard.writeText(
      stripeWhitespace ? textToCopy.replace(/\s/g, "") : textToCopy
    );

  return (
    <StyledSpan>
      <Tooltip title={tooltipTitle}>
        <ButtonBase onClick={handleClickCopyBtn}>{textToCopy}</ButtonBase>
      </Tooltip>
      <Tooltip title={tooltipTitle}>
        <IconButton onClick={handleClickCopyBtn}>
          <ContentCopyIcon />
        </IconButton>
      </Tooltip>
    </StyledSpan>
  );
};

const StyledSpan = styled("span")(({ theme: { palette } }) => ({
  display: "inline-flex",
  alignItems: "center",
  verticalAlign: "middle",

  [`& > .${buttonBaseClasses.root}:first-of-type`]: {
    fontFamily: `source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace`,
    cursor: "pointer",
    display: "inline-block",
    margin: "0 0.25ch",
    padding: "0.1rem 0.5rem",
    borderRadius: "0.25rem",
    textAlign: "center",
    verticalAlign: "middle",
    color: THEMES.DARK.palette.text.primary,
    backgroundColor:
      palette.mode === "dark"
        ? THEMES.DARK.palette.background.paper
        : alpha(THEMES.DARK.palette.background.paper, 0.8),
  },

  [`& > .${iconButtonClasses.root} > .${svgIconClasses.root}`]: {
    fontSize: "1rem",
  },
}));

export type ClickToCopyTextProps = {
  children: string;
  stripeWhitespace?: boolean;
  tooltipTitle?: string;
};
