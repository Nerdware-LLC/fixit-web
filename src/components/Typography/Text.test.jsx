import renderer from "react-test-renderer";
import { Text } from "./Text";
import { ThemeProvider } from "../../app/ThemeProvider";

it("renders correctly", () => {
  // ThemeProvider is necessary to be able to access prop.theme in styled css.
  const tree = renderer
    .create(
      <ThemeProvider>
        <Text>This is TEST text.</Text>
      </ThemeProvider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
