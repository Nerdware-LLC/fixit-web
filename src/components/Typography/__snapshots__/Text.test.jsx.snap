// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<p
  className="css-v8rbnv"
>
  This is TEST text.
</p>
`;
